mlir/include/mlir/Dialect/GPU/IR/GPUBase.td

```diff
@@ -99,6 +99,21 @@ def GPU_AddressSpaceEnum : GPU_I32Enum<
 def GPU_AddressSpaceAttr :
   GPU_I32EnumAttr<"address_space", GPU_AddressSpaceEnum>;
 
+// TODO is this different enough from address space?
+def GPU_StorageClass : I32EnumAttr<"StorageClass",
+    "The storage class to apply memory ordering constraints to.",
```

Comment:
Why not an array of `gpu.address_space`?

Reply:
Thanks for the suggestion. That's probably a good idea, with no array meaning all levels. 
I'll give that a shot tomorrow.